doc/admin-guide/files/records.config.en.rst

```diff
@@ -990,6 +990,22 @@ mptcp
    ``thread`` Re-use sessions from a per-thread pool.
    ========== =================================================================
 
+.. ts:cv:: CONFIG proxy.config.http.server_session_sharing.pool_hybrid_limit INT 0
+
+   Setting :ts:cv:`proxy.config.http.server_session_sharing.pool` to global can reduce
+   the number of connections to origin for some traffic loads.  However, if many
+   execute threads are active, the thread contention on the global pool can reduce the
+   lifetime of connections to origin and reduce effective origin connection reuse. The
+   :ts:cv:`proxy.config.http.server_session_sharing.pool_hybrid_limit` setting lets you
+   enable a hybrid pool mode using both global and per thread pools.  If hybrid mode
+   is enabled, sessons are returned to the local thread pool if the global pool lock is
+   not acquired rather than just closing the origin connection as is the case in standard
+   global mode.
+
+   Setting it to 0, disables hybrid mode.  Setting it to -1, enables hybrid
```

Comment:
This looks a bit vague. The hybrid mode is enabled or disabled regardless of the setting of `proxy.config.http.server_session_sharing.pool`?

I'd recommend below to make it clear. 
1. add `hybrid` mode to `proxy.config.http.server_session_sharing.pool`
2. `proxy.config.http.server_session_sharing.pool_hybrid_limit` works with `hybrid` mode only
3. `0` for no limit

Reply:
Yes, as it stands, it is using hybrid limit as a modifier to the pool setting.  As I noted below, it isn't clear to me from my testing so far that you gain anything by messing with the limit beyond turning it all the way on (-1) or off (0).  In that case, I agree that just making a hybrid pool option would be a clearer config.

I guess I could keep around the hybrid_limit has a lesser used tuning option that is set to -1 by default but only used if pool == hybrid.

Reply:
Agreed. Adding a setting which there is no way to find a better value for than "on or off" doesn't make sense, and does more harm. I'd say nuke this setting for now (until it's shown to have a need), and add this hybrid model to the existing setting with another numeric value.